opendatahub-io/odh-dashboard
# Delete pipeline server from the Pipelines Section in Projects page

Fixes #1542 

## Description

We can now delete pipeline server from the Pipelines Section in Projects page. Reused `PipelinesPageHeaderActions`.The functionality is the same as `Pipeline Server Actions` dropdown in Global Pipelines page, only the toggle is `KebabToggle`.

More details: #1542 
The design has been provided in #1542 


Edit - @kywalker-rh implemented the design provided in #1542 Also, what do we do if pipeline server is not installed? Currently, the "Import pipeline" button is disabled when there is no pipeline server. Do I disable this menu or hide it altogether? I'm hiding it currently. You can check the screenshots below for all states:

Edit 1: Disabling both the kebab menu on Projects page and the "Pipeline Server Actions" button in Global Pipelines page when there is no pipeline server (https://github.com/opendatahub-io/odh-dashboard/pull/1743#issuecomment-1711878609)
 
With pipeline server installed:

<img width="657" alt="Screenshot 2023-09-05 at 5 35 10 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/22885912/4dc22a37-af18-4341-9072-7b704f425da6">

With no pipeline server - Projects page(Disabled kebab menu beside "Import Pipeline" button):

<img width="977" alt="Screenshot 2023-09-11 at 11 31 43 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/22885912/f3022d42-00c4-4a81-bf9f-3c86e694087f">

With no pipeline server - Global Pipelines page(Disabled "Pipelines Server Actions" button on the top-right):

<img width="1035" alt="Screenshot 2023-09-11 at 11 32 03 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/22885912/211f2418-462b-427b-98d3-ab1641926063">

Tooltips when there is no pipeline server:

Global Pipelines page:

<img width="1031" alt="Screenshot 2023-09-26 at 10 37 14 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/22885912/0b588c31-20d6-49f8-a96e-114619819c57">

Projects page:
<img width="827" alt="Screenshot 2023-09-26 at 10 36 59 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/22885912/8b85c866-c891-42f8-9e65-16e1abbf6375">

## How Has This Been Tested?



1. Go to Projects page and scroll/navigate to the Pipelines section 
2. Check whether you can see the kebab menu in the Pipelines Section(you should only see the enabled kebab menu icon when you have a pipeline server installed and/or the pipeline is in an initializing state, and in a disabled state when there is no pipeline server) and check the functionality of each dropdown item. Check whether you can see the tooltip when the kebab menu is disabled. 
3. Go to Global Pipelines page and check whether you see the "Pipeline Server Actions" dropdown in the header page and check the functionality of dropdown items. This will be disabled when there is no Pipeline server. You can see the tooltip when there is no Pipeline Server.

## Test Impact


TBD

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
